### DISCUSSION TITLE
After compressing application tries to load assembly with wrong version

### DISCUSSION THREAD

**Discussion by kryzys69** [mkaring/ConfuserEx#458] [Q&A]
I don't know if I understand it correctly, but it looks like this:

Application has reference to:
    System.Runtime.CompilerServices.Unsafe - version 6.0.0.0

in app.config there is a binding:

<dependentAssembly>
    <assemblyIdentity name="System.Runtime.CompilerServices.Unsafe" publicKeyToken="b03f5f7f11d50a3a" culture="neutral" />
    <bindingRedirect oldVersion="0.0.0.0-6.0.0.0" newVersion="6.0.0.0" />
</dependentAssembly>

After compression (without any protection) application doesn't start with the information that it could not load previously mentioned assembly but in version 4.0.4.1. It looks like it doesn't see  binding redirect - version 6.0.0.0 should be used instead.

Thanks,
Chris

**Comment by mkaring** [comment]
There is a file produced next to the unprotected executable. Matching the name of your executable, but with the extension `.config`. You may need to copy this file next to your protected assembly. It contains the redirection instructions for the assembly.

**Comment by kryzys69** [comment]
Thanks, I didn't realize it. I have one, maybe stupid question... is it possible to make packaged, protected executable self-contained? I mean embed external configs in it?

**Reply by mkaring** [reply]
The default handling of the assembly redirects requires the config file to work correctly. You'd have to implement your own redirection logic for the assemblies into the `AssemblyResolve` events to work around it. It's not that difficult to do in the first place, but it adds additional work to maintain changing dependencies.

**Comment by kryzys69** [comment]
Thanks for the tip, I'll check it out
